WBooks/Views/AddNew/AddNewViewController.swift

```diff
@@ -8,20 +8,108 @@
+            case .default:
+                if UIImagePickerController.isSourceTypeAvailable(.photoLibrary) {
+                    let imagePicker = UIImagePickerController()
+                    imagePicker.delegate = self
+                    imagePicker.sourceType = .photoLibrary
+                    imagePicker.allowsEditing = true
+                    self.present(imagePicker, animated: true, completion: nil)
+                }
+                
+            case .cancel:
+                break
+                
+            case .destructive:
+                break
+            }
+        }))
+        alert.addAction(UIAlertAction(title: "Camera", style: .default, handler: { action in
+            switch action.style {
+            case .default:
+                if UIImagePickerController.isSourceTypeAvailable(.camera) {
+                    let imagePicker = UIImagePickerController()
+                    imagePicker.delegate = self
+                    imagePicker.sourceType = .camera
+                    imagePicker.allowsEditing = false
+                    self.present(imagePicker, animated: true, completion: nil)
+                }
+            case .cancel:
+                break
+                
+            case .destructive:
+                break
+            }
+        }))
+        self.present(alert, animated: true, completion: nil)
+//        if UIImagePickerController.isSourceTypeAvailable(.photoLibrary) {
+//            var imagePicker = UIImagePickerController()
+//            imagePicker.delegate = self
+//            imagePicker.sourceType = .photoLibrary;
+//            imagePicker.allowsEditing = true
+//            self.present(imagePicker, animated: true, completion: nil)
+//        }
+        
+    }
+    @objc func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String: AnyObject]) {
+        //swiftlint:disable:next force_cast
+        let image = info[UIImagePickerControllerOriginalImage] as! UIImage
+        _view.bookImage.image = image
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc func onTap(_ sender: Any) {
+        let image: UIImage = _view.bookImage.image!
+        let title: String = _view.name.text!
+        let author: String = _view.author.text!
+        let year: String = _view.year.text!
+        let genre: String = _view.topic.text!
+        let descript: String = _view.descript.text!
+        if title == "" || author == "" {
+            let alert = UIAlertController(title: "Error", message: "all fields must be complete",
+                                          preferredStyle: UIAlertControllerStyle.alert)
+            alert.addAction(UIAlertAction(title: "Gallery", style: .default, handler: nil))
+            self.present(alert, animated: true, completion: nil)
+            return
+        }
+        
+        RepositoryCreator.shared.bookRepository.postSuggestion(title: title, author: author)
```

Comment:
This should be inside the else clause in the previous if. As a general rule avoid early return in if statements when there is clear alternative execution path

Reply:
Early returns should be used for preconditions checks generally in the form of guard statements